Makefile

```diff
@@ -105,12 +105,21 @@ test: ## Run tests
 test-integration: ## Run integration tests
 	go test -v -tags=integration ./test/integration/...
 
-.PHONY: test-e2e
-test-e2e: ## Run e2e tests
-	PULL_POLICY=IfNotPresent $(MAKE) docker-build
+.PHONY: full-capd-e2e
+full-capd-e2e: ## Rebuild all manifests and all provider images then run the capd-e2es
 	$(MAKE) generate-manifests
 	$(MAKE) release-manifests
-	cd ./test/e2e; MANAGER_IMAGE=$(CONTROLLER_IMG)-$(ARCH):$(TAG) go test -v -tags=e2e -timeout=1h . -args -ginkgo.v -ginkgo.trace
+	$(MAKE) all-images-capd-e2e
+
+.PHONY: all-images-capd-e2e
+all-images-capd-e2e: ## Rebuild all Cluster API provider images and run the capd-e2e tests
+	make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api PULL_POLICY=IfNotPresent
+	$(MAKE) capd-e2e
+
+.PHONY: capd-e2e
+capd-e2e: ## Rebuild the docker provider and run the capd-e2e tests
```

Comment:
Just a suggestion but I think it would be more intuitive to do `make test-capd-e2e`